canal/sync.go

```diff
@@ -31,9 +31,10 @@ func (c *Canal) startSyncer() (*replication.BinlogStreamer, error) {
 		log.Infof("start sync binlog at binlog file %v", pos)
 		return s, nil
 	} else {
-		s, err := c.syncer.StartSyncGTID(gset)
+		copyedGset, _ := mysql.ParseGTIDSet(c.cfg.Flavor, gset.String())
```

Comment:
maybe we should return a copy in `c.master.GTIDSet()` directly

Reply:
you can  review  https://github.com/siddontang/go-mysql/blob/3fb4533b76294813dde21d94e80516828b0f977c/mysql/gtid.go#L17
there is a `Clone` function